### PR TITLE
fix: relation on self not working

### LINKED ISSUE #2599: fix: multiple relations on an object on the same entity
## Scope & Context

This ticket is following the new implementation of flexible back-end well described here: #1142 
It seems that we have an issue when making two relations on a same object targeting the same entity.

## Current behavior

Only one of the relations is exposed.

## Expected behavior

All the relations should be properly exposed.

### PR DESCRIPTION
**Pull request by magrinj**
Relation on self was not working, this PR fix the ticket #2599.
You can now create relation like this one `Person <-> Person`.

### CODE REVIEW

File: server/src/metadata/relation-metadata/relation-metadata.service.ts [hunk 1]
```diff
@@ -168,16 +168,16 @@ export class RelationMetadataService extends TypeOrmQueryService<RelationMetadat
...
       if (fieldMetadata.type === FieldMetadataType.RELATION) {
-        acc[fieldMetadata.objectMetadataId] = fieldMetadata;
+        acc[fieldMetadata.name] = fieldMetadata;
       }
 
       return acc;
     }, {});
 
     const createdRelationMetadata = await super.createOne({
       ...record,
```
Comment (charlesBochet) [inline server/src/metadata/relation-metadata/relation-metadata.service.ts:178]: I'm surprise about the usage of "record" here!?
it should be relation or relationMetadata

File: server/src/metadata/relation-metadata/relation-metadata.service.ts [hunk 2]
```diff
@@ -168,16 +168,16 @@ export class RelationMetadataService extends TypeOrmQueryService<RelationMetadat
 
     const createdFieldMap = createdFields.reduce((acc, fieldMetadata) => {
       if (fieldMetadata.type === FieldMetadataType.RELATION) {
-        acc[fieldMetadata.objectMetadataId] = fieldMetadata;
+        acc[fieldMetadata.name] = fieldMetadata;
```
Comment (charlesBochet) [inline server/src/metadata/relation-metadata/relation-metadata.service.ts:171]: can we pass the fieldMetadata.id instead? I feel the name is not robust enough. We could have two field with same name too

**Review by charlesBochet — COMMENTED** [review]
I have left comments!

File: server/src/metadata/relation-metadata/relation-metadata.service.ts (hunk 2 above)
Comment (magrinj) [reply server/src/metadata/relation-metadata/relation-metadata.service.ts:171]: @charlesBochet Actually we can't as we need to retrieve the fieldMetadata.id for `from` and `to` to store them inside the `relationMetadata`.
We can't base the code on the order of the declaration inside the `createMany` as Postgres doesn't guarantee the same order in the output of an `INSERT`.
Also name are unique and we can't have the same name multiple times on a same object :)

File: server/src/metadata/relation-metadata/relation-metadata.service.ts (hunk 1 above)
Comment (magrinj) [reply server/src/metadata/relation-metadata/relation-metadata.service.ts:178]: Doesn't really have preference on that as it's referencing the user `input`, in fact it should be something like `relationInput` but name `record` is used in other places in the code for inputs.
I can take a quick look and rename them ? Or maybe we can create a small ticket for that ?

Comment (charlesBochet) [reply server/src/metadata/relation-metadata/relation-metadata.service.ts:178]: let's create a small ticket for that! But yes record should be only used to describe the data itself.
The full naming on the project is: an object **record** is an instance of an object **metadata** (class)

**Review by charlesBochet — APPROVED** [review]
LGTM!

**Pull request merged**